source/Engine.cpp

```diff
@@ -2742,3 +2742,18 @@ void Engine::EmplaceStatusOverlay(const shared_ptr<Ship> &it, Preferences::Overl
 	statuses.emplace_back(it->Position() - center, it->Shields(), it->Hull(),
 		min(it->Hull(), it->DisabledHull()), max(20., width * .5), type, alpha);
 }
+
+
+
+void Engine::HandleEvents()
+{
+	set<const Ship *> disabledFighters;
+	for(const auto &event : events)
+		if(event.Type() & ShipEvent::DISABLE && event.Target()->CanBeCarried())
+			disabledFighters.insert(event.Target().get());
+	for(auto &projectile : projectiles)
+		if(disabledFighters.count(projectile.Target()))
+			projectile.BreakTarget();
```

Comment:
No use in iterating over every projectile every frame if we're not doing anything with them.
```suggestion
	if(!disabledFighters.empty())
		for(auto &projectile : projectiles)
			if(disabledFighters.count(projectile.Target()))
				projectile.BreakTarget();
```

Reply:
> Do we even have to use events? Projectiles already break their targetting in a couple of cases, so we could just add a check for disabled fighter to the conditions (in Projectiles::Move).

Wouldn't that have worse performance? That would be like iterating over every projectile every frame, which is worse than only checking the events.q

Reply:
We already have to iterate over every projectile every frame to Move() them.

Reply:
huh yeah I guess so whoops mb, misunderstood the intent.